Add tests, bump dependencies.

skylarmb opened the pull request:
- Bumps `react-script-hook` to 1.6 to hopefully solve #239 
- Adds tests to close #29 and reduce future regressions

cgfarmer4 reviewed (approved):
Yarn.lock diff seems high? Stamping to unblock.

skylarmb commented:
@cgfarmer4 i added 5 packages in order to enable running tests in this repo and updated 1 package, so i think this is expected. I will spot check the lockfile though.

added:
- `@babel/preset-typescript`
- `@testing-library/react`
- `@testing-library/react-hooks`
- `@types/jest`
- `jest`

updated:
- `react-script-hook`

skylarmb commented:
the diffs seem to be largely a switch from `sha1` to `sha512` checksums, which is expected as the default checksum has changed to `sha512` in both npm and yarn. there are also some diffs where packages now resolve to `registry.npmjs.com` instead of the old yarn registry. This is expected due to the `.npmrc` file, and any package that is installed or re-installed from now on should see a similar change in the lockfile if its using the previous (unspecified) registry instead of the registry from `.npmrc`.
